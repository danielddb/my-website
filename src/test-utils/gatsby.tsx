import { FixedObject } from 'gatsby-image';

export const createGatsbyFixedImageMock = (): FixedObject => ({
  base64:
    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z',
  width: 64,
  height: 64,
  src: '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg',
  srcSet:
    '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x'
});
