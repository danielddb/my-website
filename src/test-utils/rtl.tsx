import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import light from '../themes/light';

const Wrapper = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
