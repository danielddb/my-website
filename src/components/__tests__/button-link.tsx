import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'test-utils/rtl';
import ButtonLink from '../button-link';

test('it should render correctly', () => {
  const { container } = render(<ButtonLink to="/">Test</ButtonLink>);
  expect(container).toMatchSnapshot();
});

test('it should render correctly when rendered as a native "a" tag', () => {
  const { container } = render(
    <ButtonLink as="a" href="/">
      Test
    </ButtonLink>
  );
  expect(container).toMatchSnapshot();
});

test('it should render correctly with the primary styles applied', () => {
  const { container } = render(
    <ButtonLink as="a" href="/" primary={true}>
      Test
    </ButtonLink>
  );
  expect(container).toMatchSnapshot();
});

test('it should render with no accessibility issues', async () => {
  const { container } = render(<ButtonLink>Test</ButtonLink>);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
