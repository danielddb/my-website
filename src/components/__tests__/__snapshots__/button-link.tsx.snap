// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  display: inline-block;
  position: relative;
  padding: 1rem 2rem;
  border-radius: 0.25rem;
  cursor: pointer;
  background: none;
  color: #101010;
  border: 0.1875rem solid #101010;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  line-height: 1rem;
  font-weight: bold;
  text-align: center;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus,
.c1:hover {
  background: none;
}

.c1:active {
  background: none;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <a
    class="c0 c1"
    href="/"
  >
    Test
  </a>
</div>
`;

exports[`it should render correctly when rendered as a native "a" tag 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  display: inline-block;
  position: relative;
  padding: 1rem 2rem;
  border-radius: 0.25rem;
  cursor: pointer;
  background: none;
  color: #101010;
  border: 0.1875rem solid #101010;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  line-height: 1rem;
  font-weight: bold;
  text-align: center;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus,
.c0:hover {
  background: none;
}

.c0:active {
  background: none;
}

<div>
  <a
    class="c0"
    href="/"
  >
    Test
  </a>
</div>
`;

exports[`it should render correctly with the primary styles applied 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  display: inline-block;
  position: relative;
  padding: 1rem 2rem;
  border-radius: 0.25rem;
  cursor: pointer;
  background: #3701b3;
  color: #fff;
  border: 0.1875rem solid transparent;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  line-height: 1rem;
  font-weight: bold;
  text-align: center;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus,
.c0:hover {
  background: #4b1abb;
}

.c0:active {
  background: #3201a1;
}

<div>
  <a
    class="c0"
    href="/"
  >
    Test
  </a>
</div>
`;
